request/publishing.go

```diff
@@ -24,12 +24,19 @@ func NewPublishingRoundTripper(
 
 func (rt *PublishingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	startTime := time.Now()
+	rt.publisher.Publish(events.DataRequestAPI{
```

Comment:
This will also cause logger subscriber to print almost the same message twice. Do we want to log attempts as well?

Reply:
Good point. Will introduce an early-exit if logger is handling attempt events, so the output shouldn't get cluttered.